docs/source/docker_only.rst

```diff
@@ -0,0 +1,129 @@
+It is possible to develop with dallinger using only docker as a prerequisite.
+This guide goes through the necessary steps to achieve this.
+
+Set up services
+===============
+
+Create the dallinger docker network (if not present already):
+
+.. code-block:: shell
+
+    docker network create dallinger
+
+Start services needed by dallinger:
+
+.. code-block:: shell
+
+    docker run -d --name dallinger_redis --network=dallinger -v dallinger_redis:/data redis redis-server --appendonly yes
+    docker run -d --name dallinger_postgres --network=dallinger -e POSTGRES_USER=dallinger -e POSTGRES_PASSWORD=dallinger -e POSTGRES_DB=dallinger -v dallinger_postgres:/var/lib/postgresql/data postgres:12
+
+If you have these already created you can remove the existing ones and recreate them. To remove the existing ones run:
+
+.. code-block:: shell
+
+    # Needed if you want to recreate containers
+    docker rm dallinger_redis
+    docker rm dallinger_postgres
+
+or you can start the containers you have:
+
+.. code-block:: shell
+
+    # Needed if you want to start stopped containers
+    docker start dallinger_redis dallinger_postgres
+
+Start adminer:
+
+.. code-block:: shell
+
+    docker run -d -p 8080:8080 --name dallinger_adminer --network dallinger --link dallinger_postgres:db adminer
+
+Connect to adminer by visiting http://localhost:8080
+Select PostgreSQL from the dropdown, and enter `dallinger` as both username and password. You can leave the "database" field blank to get a list of existing databases.
+
+
+Run experiment from docker image
+================================
+
+Enter the experiment directory:
+
+.. code-block:: shell
+
+   cd /path/to/experiment-directory
+
+Create a file named `Dockerfile` with these contents (replace image name in the `FROM` directive to match the dallinger version you depend on in your `constraints.txt` file):
+
+.. code-block::
+
+    # syntax = docker/dockerfile:1.2
+    FROM ghcr.io/dallinger/dallinger:9.1.0a1
+
+    RUN mkdir /experiment
+    COPY requirements.txt /experiment/requirements.txt
+    COPY constraints.txt /experiment/constraints.txt
+
+    WORKDIR /experiment
+
+    RUN python3 -m pip install -e /dallinger[docker]
```

Comment:
I guess this should ideally go before the requirements copying, so that the cache doesn't get invalidated when the requirements change?

Reply:
Good catch. I will remove this one from here and place it in the base image (top level `Dockerfile`) so it's not needed anymore.

Reply:
Ah yes that would be great -- I thought maybe it was here because you wanted to avoid installing Docker dependencies on the base image .